packages/manager/modules/vrack/src/order/service.js

```diff
@@ -0,0 +1,55 @@
+export default class vrackOrderService {
+  /* @ngInject */
+  constructor($http, coreConfig) {
+    this.$http = $http;
+    this.user = coreConfig.getUser();
+  }
+
+  createCartWithOneVrack() {
+    let cartId;
+    return this.createCart()
+      .then(({ data: addVrackData }) => {
+        cartId = addVrackData.cartId;
+        return this.assignCart(cartId);
+      })
+      .then(() => this.addVrackToCart(cartId));
+  }
+
+  createCart() {
+    return this.$http.post('/order/cart', {
+      ovhSubsidiary: this.user.ovhSubsidiary,
+    });
+  }
+
+  addVrackToCart(cartId) {
+    return this.$http.post(`/order/cart/${cartId}/vrack`, {
+      duration: 'P1M',
+      planCode: 'vrack',
+      pricingMode: 'default',
+      quantity: 1,
+    });
+  }
```

Comment:
**suggestion:** The params can be hard-coded on the controller and received as an arguments to this function. Thus makes the function in the service can be reusable if needed.